### PR TITLE
Clarify deserialization to object behavior

### LINKED ISSUE #16859: Serializer behavior of deserializing objects as JsonElement instances applies to objects anywhere in a type graph
The documentation states "System.Text.Json stores a boxed JsonElement for both primitive and complex values within the System.Object property or dictionary value. "

This behavior doesn’t apply only to POCO properties or dictionary values, as the migration guide states, but to objects anywhere in a given object graph (i.e also to root objects and array/list object elements. The documentation should mention this explicitly.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 6b74d77c-33e1-d44c-18c6-bd0b612f74de
* Version Independent ID: 9c6a7135-eb75-21a9-1d27-0880459523ca
* Content: [Migrate from Newtonsoft.Json to System.Text.Json - .NET](https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to#feedback)
* Content Source: [docs/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to.md](https://github.com/dotnet/docs/blob/master/docs/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to.md)
* Product: **dotnet**
* GitHub Login: @tdykstra
* Microsoft Alias: **tdykstra**

### PR DESCRIPTION
**Pull request by tdykstra**
Fixes #16859

### CODE REVIEW

File: docs/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to.md [hunk 1]
```diff
@@ -220,7 +220,14 @@ When `Newtonsoft.Json` deserializes to `object` properties in POCOs or in dictio
 * Returns a `JObject` or `JArray` for complex values in the JSON payload. *Complex values* are collections of JSON key-value pairs within braces (`{}`) or lists of values within brackets (`[]`). The properties and values within the braces or brackets can have additional properties or values.
 * Returns a null reference when the payload has the `null` JSON literal.
 
-<xref:System.Text.Json> stores a boxed `JsonElement` for both primitive and complex values within the `System.Object` property or dictionary value. However, it treats `null` the same as `Newtonsoft.Json` and returns a null reference when the payload has the `null` JSON literal in it.
+<xref:System.Text.Json> stores a boxed `JsonElement` for both primitive and complex values whenever deserializing to <xref:System.Object>, for example:
```
Comment (layomia) [inline docs/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to.md]: L217 needs to be updated as well.

**Review by layomia — APPROVED** [review]
Thank you

**Review by Thraka — APPROVED** [review]
LGTM

File: docs/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to.md [hunk 2]
```diff
@@ -214,13 +214,20 @@ To support polymorphic deserialization, create a converter like the example in [
...
+When `Newtonsoft.Json` deserializes to <xref:System.Object>, it:
 
 * Infers the type of primitive values in the JSON payload (other than `null`) and returns the stored `string`, `long`, `double`, `boolean`, or `DateTime` as a boxed object. *Primitive values* are single JSON values such as a JSON number, string, `true`, `false`, or `null`.
 * Returns a `JObject` or `JArray` for complex values in the JSON payload. *Complex values* are collections of JSON key-value pairs within braces (`{}`) or lists of values within brackets (`[]`). The properties and values within the braces or brackets can have additional properties or values.
 * Returns a null reference when the payload has the `null` JSON literal.
 
-<xref:System.Text.Json> stores a boxed `JsonElement` for both primitive and complex values within the `System.Object` property or dictionary value. However, it treats `null` the same as `Newtonsoft.Json` and returns a null reference when the payload has the `null` JSON literal in it.
+<xref:System.Text.Json> stores a boxed `JsonElement` for both primitive and complex values whenever deserializing to <xref:System.Object>, for example:
+
+* An `object` property
```
Comment (Thraka) [inline docs/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to.md]: I think these should have `.` at the end.

Comment (tdykstra) [reply docs/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to.md]: See https://styleguides.azurewebsites.net/Styleguide/Read?id=2700&topicid=28751

>If all list elements are short phrases (three words or fewer), don’t end them with periods, even if they form a complete sentence together with the list introduction. 

This is an edge case, as 2 bullets are 3 words and 2 are 4 words. But strictly speaking it doesn't match the rule.

**Pull request merged**